packages/ckeditor5-watchdog/src/editorwatchdog.ts

```diff
@@ -284,14 +306,30 @@ export default class EditorWatchdog<TEditor extends Editor = Editor> extends Wat
 	}
 
 	/**
-	 * Returns the editor data.
+	 * Gets the all data that are required to reinitialize editor instance.
 	 */
-	private _getData(): Record<string, string> {
-		const data: Record<string, string> = {};
+	private _getData(): EditorData {
+		const editor = this.editor!;
+		const roots = editor.model.document.getRootNames();
+		const data: EditorData = {};
+
+		roots.forEach( root => {
+			const markersData = {} as any;
+
+			for ( const marker of editor.model.markers ) {
+				markersData[ marker.name ] = {
+					range: marker.getRange(),
+					usingOperation: marker._managedUsingOperations,
+					affectsData: marker._affectsData
+				};
+			}
 
-		for ( const rootName of this._editor!.model.document.getRootNames() ) {
-			data[ rootName ] = this._editor!.data.get( { rootName } );
-		}
+			data[ root ] = {
+				content: JSON.stringify( Array.from( editor.model.document.getRoot( root )!.getChildren() ) ),
+				attributes: Array.from( editor.model.document.getRoot( root )!.getAttributes() ),
```

Comment:
```suggestion
				content: JSON.stringify( Array.from( root.getChildren() ) ),
				attributes: Array.from( root.getAttributes() ),
```

Reply:
I will this comment even though I noticed that you "roots" is actually "root names". The name for the variable is incorrect. I leave it to show you that it is important to correctly name stuff. My first reaction was that you have an error somewhere.

BTW. `Document#getRoots()` will be available very soon (hopefully in this release as well).

Reply:
I forgot to refactor it, sorry